crates/re_arrow_store/benches/arrow2.rs

```diff
@@ -0,0 +1,229 @@
+//! Keeping track of performance issues/regressions in `arrow2` that directly affect us.
+
+#[global_allocator]
+static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
+
+use std::sync::Arc;
+
+use arrow2::array::{Array, PrimitiveArray, StructArray};
+use criterion::{criterion_group, criterion_main, Criterion};
+use itertools::Itertools;
+use re_log_types::{
+    component_types::{InstanceKey, Point2D},
+    datagen::{build_some_instances, build_some_point2d},
+    DataCell,
+};
+
+// ---
+
+criterion_group!(benches, estimated_size_bytes);
+criterion_main!(benches);
+
+// ---
+
+#[cfg(not(debug_assertions))]
+const NUM_ROWS: usize = 10_000;
+#[cfg(not(debug_assertions))]
+const NUM_INSTANCES: usize = 100;
+
+// `cargo test` also runs the benchmark setup code, so make sure they run quickly:
+#[cfg(debug_assertions)]
+const NUM_ROWS: usize = 1;
+#[cfg(debug_assertions)]
+const NUM_INSTANCES: usize = 1;
+
+// ---
+
+fn estimated_size_bytes(c: &mut Criterion) {
+    let kind = ["primitive", "struct"];
```

Comment:
y'know an enum doesn't hurt even if its throw away, would take as much space as all these unrechables :P

Reply:
I was tired and desperate lol; yeah let's make it into an enum